### PR TITLE
perl: fix build on 10.11 with Xcode 8

### PR DESCRIPTION
**Pull request by ilovezfs**

### CODE REVIEW

**Comment by ilovezfs** [comment]
CC @mistydemeo seems upstream took your patch and ran with it, but thinks everything is fixed now https://rt.perl.org/Public/Bug/Display.html?id=128427#txn-1422265

File: Formula/perl.rb [hunk 1]
```diff
@@ -1,12 +1,25 @@
 class Perl < Formula
   desc "Highly capable, feature-rich programming language"
   homepage "https://www.perl.org/"
-  url "http://www.cpan.org/src/5.0/perl-5.24.0.tar.xz"
-  mirror "https://mirrors.ocf.berkeley.edu/debian/pool/main/p/perl/perl_5.24.0.orig.tar.xz"
-  sha256 "a9a37c0860380ecd7b23aa06d61c20fc5bc6d95198029f3684c44a9d7e2952f2"
   revision 1
 
-  head "https://perl5.git.perl.org/perl.git", branch: "blead"
+  head "git://perl5.git.perl.org/perl.git", branch: "blead"
```
Comment (DomT4) [inline Formula/perl.rb]: Why the protocol downgrade?

Comment (ilovezfs) [inline Formula/perl.rb]: it wouldn't clone for me otherwise ... does it for u?

Comment (DomT4) [inline Formula/perl.rb]: Argh, no. Looks like they've lost the `https://`, only `http://` and `git://` are listed upstream now, and only the latter works. Nice.

Comment (ilovezfs) [inline Formula/perl.rb]: Yeah, pretty gross. Maybe there's a secure mirror that's actually kept in sync ... :(

Comment (DomT4) [inline Formula/perl.rb]: Kind of tempting to just drop `HEAD`, given we use it as a `default_formula` now and Perl does not remain completely compatible across major releases.

Comment (ilovezfs) [inline Formula/perl.rb]: I basically consider head specs to be for our own internal use hehe

File: Formula/perl.rb [hunk 2]
```diff
@@ -20,17 +33,17 @@ class Perl < Formula
...
 
-  # Fixes Time::HiRes module bug related to the presence of clock_gettime
-  # https://rt.perl.org/Public/Bug/Display.html?id=128427
-  # Merged upstream, should be in the next release.
-  if MacOS.version >= :sierra
-    patch do
-      url "https://raw.githubusercontent.com/Homebrew/formula-patches/b18137128c4e0cb7e92e9ee007a9f78bc9d03b21/perl/clock_gettime.patch"
-      sha256 "612825c24ed19d6fa255bb42af59dff46ee65c16ea77abf4a59b754aa8ab05ac"
+  def install
+    if MacOS.version == "10.11" && MacOS::Xcode.installed? && MacOS::Xcode.version >= "8.0"
```
Comment (DomT4) [inline Formula/perl.rb]: Has this fix been reported upstream as well?

Comment (ilovezfs) [inline Formula/perl.rb]: most assuredly not

Comment (chdiza) [inline Formula/perl.rb]: I don't feel qualified to do it.  I don't have the ability to explain the problem or what the fix does.  Sorry! :smile:

Comment (DomT4) [inline Formula/perl.rb]: If we can at least report the issue upstream I'm a 👍 on carrying this.

Comment (DomT4) [inline Formula/perl.rb]: No preference on who does that!

**Comment by ilovezfs** [comment]
original fix:
```
      inreplace "dist/Time-HiRes/HiRes.xs" do |s|
        s.gsub! "static int clock_gettime(int clock_id, struct timespec *ts) {",
                "int clock_gettime(clockid_t clock_id, struct timespec *ts) {"

        s.gsub! "static int clock_getres(int clock_id, struct timespec *ts) {",
                "int clock_getres(clockid_t clock_id, struct timespec *ts) {"
```

**Comment by ilovezfs** [comment]
@chdiza do you mind verifying this new fix works for you? This should be a safer work around.

**Comment by chdiza** [comment]
Seems to work over here.  I'll be back if I run into some weirdness.

**Comment by ilovezfs** [comment]
Thanks @chdiza

**Pull request closed without merging**